environment/src/main/java/com/sequenceiq/environment/environment/service/EnvironmentCreationService.java

```diff
@@ -57,35 +57,39 @@ public EnvironmentCreationService(
     }
 
     // TODO: trigger creation properly
-    public void triggerCreationFlow(long envId, String envName) {
+    public void triggerCreationFlow(long envId, String envName, String accountId) {
         EnvCreationEvent envCreationEvent = EnvCreationEvent.EnvCreationEventBuilder.anEnvCreationEvent()
                 .withSelector(START_NETWORK_CREATION_EVENT.selector())
                 .withResourceId(envId)
                 .withResourceName(envName)
                 .build();
         String userCrn = authenticatedUserService.getUserCrn();
+        if (userCrn == null) {
+            userCrn = environmentService.getByName(envName, accountId).getCreator();
```

Comment:
The accound id will be null in this case as the `threadBasedUserCrnProvider` service calculate the account id from the user CRN. -> The calculation is one layer upper.